crates/primitives/transactions/src/execution.rs

```diff
@@ -575,6 +587,85 @@ mod simulate_tx_offset {
         );
     }
 
-    // TODO: add test that check that each get_account_transaction_context impl correctly uses this
-    // offset
+    #[test]
+    fn l1_handler_transaction_correctly_applies_simulate_tx_version_offset() {
+        let l1_handler_tx = L1HandlerTransaction {
+            tx: Default::default(),
+            paid_fee_on_l1: Default::default(),
+            tx_hash: Default::default(),
+        };
+
+        let original_version = l1_handler_tx.tx.version;
+        let queried_version = l1_handler_tx.get_account_transaction_context(true).version;
+
+        assert_eq!(
+            queried_version,
+            Felt252Wrapper(Felt252Wrapper::from(original_version.0).0 + SIMULATE_TX_VERSION_OFFSET).into()
+        );
+
+        let non_queried_version = l1_handler_tx.get_account_transaction_context(false).version;
+        assert_eq!(non_queried_version, original_version);
+    }
+
+    #[test]
+    fn deploy_account_transaction_correctly_applies_simulate_tx_version_offset() {
+        let deploy_account_tx = DeployAccountTransaction {
+            tx: Default::default(),
+            tx_hash: Default::default(),
+            contract_address: Default::default(),
+        };
+
+        let original_version = deploy_account_tx.tx.version;
+
+        let queried_version = deploy_account_tx.get_account_transaction_context(true).version;
+        assert_eq!(
+            queried_version,
+            Felt252Wrapper(Felt252Wrapper::from(original_version.0).0 + SIMULATE_TX_VERSION_OFFSET).into()
+        );
+
+        let non_queried_version = deploy_account_tx.get_account_transaction_context(false).version;
+        assert_eq!(non_queried_version, original_version);
+    }
+
+    #[test]
+    fn declare_transaction_correctly_applies_simulate_tx_version_offset() {
+        let declare_tx_v0 = DeclareTransaction::new(
+            starknet_api::transaction::DeclareTransaction::V0(Default::default()),
+            Default::default(),
+            ContractClass::V0(Default::default()),
+        )
+        .unwrap();
+
+        // gen TxVersion from v0 manually
+        let original_version_v0 = TransactionVersion(StarkFelt::from(0u8));
+
+        let queried_version = declare_tx_v0.get_account_transaction_context(true).version;
+        assert_eq!(
+            queried_version,
+            Felt252Wrapper(Felt252Wrapper::from(original_version_v0.0).0 + SIMULATE_TX_VERSION_OFFSET).into()
+        );
+
+        let non_queried_version = declare_tx_v0.get_account_transaction_context(false).version;
+        assert_eq!(non_queried_version, original_version_v0);
+    }
+
+    #[test]
+    fn invoke_transaction_correctly_applies_simulate_tx_version_offset() {
+        let declare_tx_v0 = InvokeTransaction {
```

Comment:
var `declare_tx_v0` is of type `InvokeTransaction`

Reply:
Slipped through somehow. Fixed